File: src/main/java/com/example/aidemo/service/NoteService.java
package com.example.aidemo.service;

import com.example.aidemo.model.Note;
import com.example.aidemo.repository.NoteRepository;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;

@Service
@RequiredArgsConstructor
public class NoteService {

    private final NoteRepository noteRepository;

    public List<Note> getAllNotes() {
        return noteRepository.findAll();
    }

    public Note getNoteById(Long id) {
        return noteRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Note not found with id: " + id));
    }

    public List<Note> searchNotes(String query) {
        List<Note> titleResults = noteRepository.findByTitleContainingIgnoreCase(query);
        List<Note> contentResults = noteRepository.findByContentContainingIgnoreCase(query);
        
        // Combine results and remove duplicates
        titleResults.removeAll(contentResults);
        titleResults.addAll(contentResults);
        
        return titleResults;
    }

    public Note createNote(Note note) {
        Date now = new Date();
        note.setCreatedAt(now);
        note.setUpdatedAt(now);
        return noteRepository.save(note);
    }

    public Note updateNote(Long id, Note noteDetails) {
        Note note = getNoteById(id);
        
        note.setTitle(noteDetails.getTitle());
        note.setContent(noteDetails.getContent());
        note.setUpdatedAt(new Date());
        
        return noteRepository.save(note);
    }

    public void deleteNote(Long id) {
        Note note = getNoteById(id);
        noteRepository.delete(note);
    }
}

File: frontend/src/services/NoteService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/notes';

class NoteService {
  getAllNotes() {
    return axios.get(API_BASE_URL);
  }

  getNoteById(noteId) {
    return axios.get(`${API_BASE_URL}/${noteId}`);
  }

  createNote(note) {
    return axios.post(API_BASE_URL, note);
  }

  updateNote(noteId, note) {
    return axios.put(`${API_BASE_URL}/${noteId}`, note);
  }

  deleteNote(noteId) {
    return axios.delete(`${API_BASE_URL}/${noteId}`);
  }

  searchNotes(query) {
    return axios.get(`${API_BASE_URL}/search?query=${query}`);
  }
}

export default new NoteService();